src/VirtualClient/VirtualClient.Contracts/ExecutionProfile.cs

```diff
@@ -62,6 +62,20 @@ public ExecutionProfile(
                 ? new Dictionary<string, IConvertible>(parameters, StringComparer.OrdinalIgnoreCase)
                 : new Dictionary<string, IConvertible>(StringComparer.OrdinalIgnoreCase);
 
+            if (this.Actions?.Any() == true)
```

Comment:
Why do we need this? Didn't find where this is needed in this PR. Also would it be better to put this in RunProfileCommand so it's not duplicated this and yaml shim?

Reply:
This is going to be needed to allow handling of scenarios in Components like the ExecuteCommand when we are using it as either a Dependency or an Action. We need to know when throwing exceptions (e.g. process.ThrowOnDependencyFailed() or process.ThrowOnWorkloadFailed()). Failures of Dependencies are handled differently than failures of Actions or Monitors. We MIGHT want to just simplify the ThrowOn* stuff in the future to NOT make a distinction on DependencyException or WorkloadException etc... but today we need to know where the Component exists in the profile.